scripts/build/autotools/CheckSSL.m4

```diff
@@ -0,0 +1,240 @@
+PHP_ARG_WITH([mongodb-ssl],
+             [whether to enable crypto and TLS],
+             [AS_HELP_STRING([--with-mongodb-ssl=@<:@auto/no/openssl/libressl/darwin@:>@],
+                             [Enable TLS connections and SCRAM-SHA-1 authentication [default=auto]])],
+             [auto],
+             [no])
+
+PHP_ARG_ENABLE([mongodb-crypto-system-profile],
+               [whether to use system crypto profile],
+               [AC_HELP_STRING([--enable-mongodb-crypto-system-profile],
+                               [Use system crypto profile (OpenSSL only) [default=no]])],
+               [no],
+               [no])
+
+PHP_ARG_WITH([openssl-dir],
+             [OpenSSL library path (deprecated)],
+             [AC_HELP_STRING([--with-openssl-dir=@<:@DIR@:>@],
+                             [OpenSSL library path (deprecated for pkg-config)])],
+             [yes],
+             [no])
+
+PHP_ARG_WITH([system-ciphers],
+             [whether to use system crypto profile (deprecated)],
+             AC_HELP_STRING([--enable-system-ciphers],
+                            [whether to use system crypto profile (deprecated for --enable-mongodb-crypto-system-profile) [default=no]]),
+             [no],
+             [no])
+
+AS_IF([test "$PHP_MONGODB_SSL" = "openssl" -o "$PHP_MONGODB_SSL" = "auto"],[
+  found_openssl="no"
+
+  PKG_CHECK_MODULES([PHP_MONGODB_SSL],[openssl],[
+    dnl PKG_CHECK_MODULES() doesn't check for headers, so confirm they exist (mainly for macOS)
```

Comment:
That seems odd. The whole idea behind pkg_config  is that its --includes (or whatever the flag is), shows all the header locations. I don't think we should go check all the headers manually where pkg-config works.